src/main/java/com/splunk/logging/HttpEventCollectorSender.java

```diff
@@ -121,28 +124,43 @@ public HttpEventCollectorSender(
           String sendModeStr,
           boolean ack,
           String ackUrl,
+          String healthUrl,
           Map<String, String> metadata) {
     this.url = Url;
     this.token = token;
     this.ack = ack;
     this.ackUrl = ackUrl;
-        if (maxEventsBatchCount == 0 && maxEventsBatchSize > 0) {
+    if (maxEventsBatchCount == 0 && maxEventsBatchSize > 0) {
       this.maxEventsBatchCount = Long.MAX_VALUE;
     } else if (maxEventsBatchSize == 0 && maxEventsBatchCount > 0) {
       this.maxEventsBatchSize = Long.MAX_VALUE;
     }
    this.delay= delay; 
    this.metadata=metadata;
+   this.healthUrl = healthUrl;
 
     if (ack) {
-      if (null == ackUrl || ackUrl.isEmpty()) {
+      if (null == ackUrl || ackUrl.isEmpty() ) {
         throw new RuntimeException(
                 "AckUrl was not specified, but HttpEventCollectorSender set to use acks.");
       }
       this.ackMiddleware = new AckMiddleware(this);
       this.middleware.add(ackMiddleware);
     }
-    
+
```

Comment:
health polling should only be performed if(ack). Little bit of a moot point, since we will eventually dump the other functionality of this logging framework. But as long as it's in the logging framework, don't start polling for health unless we are using acks.